hera_qm/omnical_metrics.py

```diff
@@ -1,14 +1,872 @@
+
+    outpath : str, default=None
+        path to place file in
+        will default to location of *omni.calfits file
+    """
+    custom_ax = True
+    if ax is None:
+        custom_ax = False
+        fig = plt.figure(figsize=(8, 6))
+
+        if ax is None:
+            ax = fig.add_subplot(111)
+
+    # get y data
+    yloc = metrics['chisq_ant_std_loc']
+    ysig = metrics['chisq_ant_std_scale']
+    ycut = yloc + metrics['chisq_std_cut'] * ysig
+    chisq_ant_std = np.array(metrics['chisq_ant_std'].values())
+    ymax = np.max([chisq_ant_std.max() * 1.2, ycut + ysig])
+
+    # make grid and plots
+    ax.grid(True)
+    p1 = ax.axhspan(yloc - ysig, yloc + ysig,
+                    color='green', alpha=0.1)
+    p2 = ax.axhline(yloc, color='steelblue')
+    p3 = ax.axhline(ycut, color='darkred')
+    p4 = ax.axhspan(ycut, ymax, color='darkred', alpha=0.2)
+    p5 = ax.plot(chisq_ant_std, marker='o', color='k', linestyle='', markersize=6)
+    ax.legend([p2, p1, p3], ["avg std", "1-sigma", "std cut"], fontsize=12)
+
+    # adjust axes
+    ax.set_xlabel('antenna number', fontsize=14)
+    ax.set_ylabel('chi-square stand dev across file', fontsize=14)
+    ax.tick_params(size=8)
+    ax.set_xticks(range(metrics['Nants']))
+    ax.set_xticklabels(metrics['ant_array'])
+    [t.set_rotation(20) for t in ax.get_xticklabels()]
+    ax.set_ylim(0, ymax)
+    ax.set_title("{0}".format(metrics['filename']))
+
+    if save is True and custom_ax is False:
+        if fname is None:
+            fname = metrics['filename'] + '.chisq_std.png'
+        if outpath is None:
+            fname = os.path.join(metrics['filedir'], fname)
+        else:
+            fname = os.path.join(outpath, fname)
+        fig.savefig(fname, bbox_inches='tight')
+
+
+class OmniCal_Metrics(object):
+    """
+    OmniCal_Metrics class for holding omnical data
+    and running metrics on them.
+    """
+
+    jones2pol = {-5: 'XX', -6: 'YY', -7: 'XY', -8: 'YX'}
+
+    def __init__(self, omni_calfits, history=''):
+        """
+        Omnical Metrics initialization
+
+        Input:
+        ------
+        omni_calfits : str
+            calfits file output from omnical, typically
+            ending in *.omni.calfits
+
+        history : str
+            history string
+        """
+        # Get file info and other relevant metadata
+        self.filedir = os.path.dirname(omni_calfits)
+        self.filestem = '.'.join(os.path.basename(omni_calfits).split('.')[:-1])
+        self.filename = os.path.basename(omni_calfits)
+        self.version_str = hera_qm_version_str
+        self.history = history
+        self.firstcal_file = None
+
+        # Instantiate Data Object
+        self.uv = UVCal()
+        self.uv.read_calfits(omni_calfits)
+
+        # Get relevant metadata
+        self.Nants = self.uv.Nants_data
+        self.freqs = self.uv.freq_array.squeeze()
+        self.Nfreqs = len(self.freqs)
+        self.jones = self.uv.jones_array[0]
```

Comment:
Is there an implicit assumption that only one polarization is present? We made this assumption for firstcal_metrics because we could get away with it in the RTP. But omnical files have multiple pols, and I don't think this will work.
Also, the M&C needs the pols to be antenna feed pols (single character `'x'` or `'y'`). We can work around this if there is need to include cross pol metrics, but my guess is that it's more accurate to keep it single.

Reply:
yes, I guess that was how I interpreted your previous comment on keeping things separated, which I took to mean run the metrics on a file-per-file basis, in which case only one pol is present. let me know if you'd like me to switch back to the multi-pol version I had before, if you mean something entirely different.

Reply:
Sorry for the confusion. The omnical files that the RTP produces have multiple pols in them. So there really isn't an option to run on a single pol file. I meant that we need to either make the module handle multiple pols - retaining which pol the metric is coming from so we can pass that on to the m&c, or if the metrics combine pols, we need to repeat the entry in the m&c so both antenna feeds show up with the same metric (ant_metrics does this for the cross pol metrics). 
One other solution would be to have a pol argument that the user supplies, then the module only uses that single pol. Then the script would be run twice - once with `'x'` and once with `'y'`.

Reply:
ah, so what you are saying is that the RTP does not produce a `zen.JD.JD.xx.HH.uvc.omni.calfits` file but a `zen.JD.JD.HH.uvc.omni.calfits` file with all 4-pol in one file? If that's the case then I can go back to the multi-pol version I had, and I'll try and make sure the relevant metadata gets propagated to M&C

Reply:
Right - just one file with all pols. Well, I think right now it's just a two-pol file (xx and yy), but that may change depending on the result of an ongoing discussion of the output of omnical ( @plaplant can correct me if I'm wrong).

Reply:
@adampbeardsley so i'm not entirely sure how to cleanly take a cross pol (i.e. 'xy' or 'yx') chisq and assign it to a single pol 'x' or 'y' per antenna, so i'm thinking of only using autopol (i.e. 'xx' or 'yy') to make the per-antenna metrics (like 'ant_chisq_avg' or 'ant_chisq_std' etc). do you have any thoughts on this?

Reply:
I think that's a fair thing to do.